cli/loader_lib.c

```diff
@@ -1,40 +1,36 @@
+// This file is a part of Julia. License is MIT: https://julialang.org/license
+//
+// This file defines an RPATH-style relative path loader for all platforms
 #include "loader.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+/* Bring in definitions of symbols exported from libjulia. */
+#include "jl_exports.h"
+
 /* Bring in helper functions for windows without libgcc. */
 #ifdef _OS_WINDOWS_
 #include "loader_win_utils.c"
 #endif
 
-/*
- * DEP_LIBS is our list of dependent libraries that must be loaded before `libjulia`.
- * Note that order matters, as each entry will be opened in-order.  We define here a
- * dummy value just so this file compiles on its own, and also so that developers can
- * see what this value should look like.  Note that the last entry must always be
- * `libjulia`, and that all paths should be relative to this loader `.exe` path.
- */
-#if !defined(DEP_LIBS)
-#define DEP_LIBS "../lib/example.so:../lib/libjulia.so"
-#endif
-static char dep_libs[256] = DEP_LIBS;
+// Save DEP_LIBS to a variable that is explicitly sized for expansion
+static char dep_libs[512] = DEP_LIBS;
```

Comment:
I don't see a way we can avoid mutating later; we need to be able to change these paths because the relative paths change between `make` time and `make install` time.

Reply:
Ah, I see, you're talking about the mutation later in this same file.  I can try to replace the `:` with `\0`, we'll see how it interacts with the rest of the processing of `DEPS_LIBS`, I have a feeling `stringreplace` isn't going to like this.

Reply:
Yeah, `stringreplace` doesn't want to deal with strings that have NULLs in them.  I'm going to leave this as-is for now.